frontend/src/app/shared/components/dataset-inputs.decorator.ts

```diff
@@ -0,0 +1,100 @@
+import 'reflect-metadata';
+import {
+  Component,
+  ElementRef,
+  ɵDirectiveDef,
+} from '@angular/core';
+
+/** We expect an ElementRef to be present on the target class */
+export interface DatasetInputsDecorated extends Component {
+  elementRef:ElementRef<HTMLElement>;
+}
+
+export interface DatasetInputsComponent extends DatasetInputsDecorated {
+  elementRef:ElementRef<HTMLElement>;
+  [key:string]:unknown;
+}
+
+/**
+ * The DatasetInputs decorator automatically sets input values from `data` attributes set on a component tag.
+ * This is useful if you're rendering the tag in the backend but want to provide data to the component via its inputs.
+ *
+ * Usage:
+ *
+ * ```
+ * @DatasetInputs
+ * @Component({ selector: 'my-component' })
+ * export class MyComponent {
+ *   @Input() someInput:string[] = [];
+ * }
+ * ```
+ *
+ * Now you can send data from the backend to the component by JSON:
+ *
+ * <%= content_tag :my_component,
+ *                 data: {
+ *                   someInput: ['a', 'b'].to_json
+ *                 }
+ *
```

Comment:
This example does not work as it is in here, it needs a few amendments:

- the closing `%>` is missing
- within the data attribute, the key must be `some-input` instead of `someInput` to result in the correct `someInput` camelCase key in the DOM map.
- `:my_component` does not work as a `content_tag`, but must be `'my-component'`

With those changes I got it to work on my branch for the query select component.

Reply:
fixed